contracts/transfer/src/wasm/bridge.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::{ops, Transfer};
+
+use canonical::{
+    BridgeStore as BridgeStoreCanon, ByteSink, ByteSource, Canon, Id32, Store,
+};
+use dusk_abi::{ContractState, ReturnValue};
+
+const PAGE_SIZE: usize = 1024 * 8;
+
+type BridgeStore = BridgeStoreCanon<Id32>;
+
+#[no_mangle]
+fn q(bytes: &mut [u8; PAGE_SIZE]) {
+    if let Err(_) = query(bytes) {
+        // TODO handle error
+    }
+}
+
+#[no_mangle]
+fn t(bytes: &mut [u8; PAGE_SIZE]) {
+    if let Err(_) = transaction(bytes) {
+        // TODO handle error
+    }
+}
+
+fn query(
+    bytes: &mut [u8; PAGE_SIZE],
+) -> Result<(), <BridgeStore as Store>::Error> {
+    let bridge = BridgeStore::default();
+    let mut source = ByteSource::new(&bytes[..], &bridge);
+    let contract: Transfer<BridgeStore> = Canon::read(&mut source)?;
+
+    let query = Canon::read(&mut source)?;
+    let ret = match query {
+        ops::QR_BALANCE => {
+            let address = Canon::read(&mut source)?;
+
+            let ret = contract.balance(address);
+            ReturnValue::from_canon(&ret, &bridge)?
+        }
+
+        ops::QR_ROOT => {
+            let ret = contract.root();
+            ReturnValue::from_canon(&ret, &bridge)?
+        }
+
+        ops::QR_NOTES_FROM_HEIGHT => {
+            let block_height = Canon::read(&mut source)?;
+
+            let ret = contract.notes_from_height(block_height);
+            ReturnValue::from_canon(&ret, &bridge)?
+        }
+
+        _ => ReturnValue::from_canon(&(), &bridge)?, /* TODO report
+                                                      * unexpected ID */
```

Comment:
File a tech-debt for that please.

Reply:
https://github.com/dusk-network/rusk/issues/193